src/knex-store.ts

```diff
@@ -44,145 +35,92 @@ function knex_store(this: any, options: Options) {
     // use in seneca.use(), eg seneca.use('knex-store').
     name: STORE_NAME,
 
-    save: asyncMethod(async function (this: any, msg: any, meta: any){
-      const seneca = this
-      
-      //Further implementation
-      // const ctx = {}
-      const ctx = intern.buildCtx(seneca, msg, meta)
-      
-      return intern.withDbClient(dbPool, ctx, async (client: any) => {
-          const ctx = { seneca, client }
-          const { ent, q } = msg
-
-          // Create a new entity
-          async function do_create() {
-            // create a new entity
-            try {
-              const newEnt = ent.clone$()
-
-              if (ent.id$){
-                newEnt.id = ent.id$
-              }
-              
-              const doCreate = await intern.insertKnex(newEnt, ctx)
-              
-              return doCreate
-              
-            } catch (err) {
-              return err
+    save: asyncMethod(async function (this: any, msg: any) {
```

Comment:
remove asyncMethod - this has no real benefit in TS, and may break with future versions of Seneca - just set the function directly

Reply:
Got it